client/src/store/map.ts

```diff
@@ -0,0 +1,45 @@
+import { atom } from "jotai";
+import { useQueryState } from "next-usequerystate";
+
+import { bboxParser, datasetsParser, layersParser, layersSettingsParser } from "./parsers";
+
+export const useSyncDatasets = () => {
+  return useQueryState("datasets", datasetsParser);
+};
+
+export const useSyncLayers = () => {
+  return useQueryState("layers", layersParser);
+};
+
+export const useSyncLayersSettings = () => {
+  return useQueryState("layers-settings", layersSettingsParser);
+};
+
+export const useSyncBbox = () => {
+  return useQueryState("bbox", bboxParser);
+};
+
+export const useSyncSearchParams = () => {
```

Comment:
There is a new way of getting the urlSearchParams inside nuqs. Check this file => 

https://github.com/Vizzuality/amazonia-360/blob/develop/client/src/app/store.ts#L32-L46

https://github.com/47ng/nuqs?tab=readme-ov-file#serializer-helper